src/lib.rs

```diff
@@ -190,11 +190,19 @@ impl Server {
 }
 
 fn write_response(response: Response<&[u8]>, mut stream: TcpStream) -> Result<(), Error> {
+    let headers = response.headers().iter().fold(
```

Comment:
i think that this might produce a malformed response if `headers` ends up blank! if you can add something to address that condition i think this looks good to go. lastly, if you wouldn't mind adding a test (or 2 - 1 for the normal, 1 for no headers) that'd be great! thanks for the contribution!

(also if you want help with the tests or feel stuck there, feel free to reply and we can add tests to this PR from our end (our setup isn't great for it at the moment, but if you're up for the challenge go for it))

Reply:
Indeed, it's causes an error with empty headers, adding a CRLF in the format string fixes it:

```
 HTTP-message   = start-line
                      *( header-field CRLF )
                      CRLF
                      [ message-body ]
```

Reply:
As per the tests, I could use some help, as I don't know yet how to mock the `TcpStream` expected by `write_response`

Reply:
ok @divarvel ! to close this out for the moment, perhaps just write an example? we'll use the example compiling for the time being :)

Reply:
Hmm, I've just changed the `TcpStream` in `write_response` to a `T: Write`, that should allow me to write a few tests.

Reply:
I've added two inline tests (since `write_response` is not public). I can move them out and make the method public.